spring-integration-file/src/main/java/org/springframework/integration/file/remote/AbstractRemoteFileStreamingMessageSource.java

```diff
@@ -162,11 +162,16 @@ public void start() {
 	@Override
 	public void stop() {
 		if (this.running.compareAndSet(true, false)) {
-			// remove unprocessed files from the queue (and filter)
-			AbstractFileInfo<F> file = this.toBeReceived.poll();
-			while (file != null) {
-				resetFilterIfNecessary(file);
-				file = this.toBeReceived.poll();
+			if (this.filter == null || this.filter.supportsSingleFileFiltering()) {
```

Comment:
Please, consider to fix unused import for `org.springframework.beans.factory.InitializingBean` in this class.
See the latest Sonar report.

Thanks